docs/proposals/job_controller.md

```diff
@@ -0,0 +1,226 @@
+**Abstract**
+
+First basic implementation proposal for a Job controller. Several exiting issues were already created regarding that particular subject:
+
+    Distributed CRON jobs in k8s #2156
+    Job Controller #1624
+
+Several features also already exist that could be used by external tools to trigger batch execution on one pod within k8 cluster.
+
+    Create a standalone pod (not linked to any existing replication controllers)
+    Execute a new process within a given container from a running pod.
+
+**Motivation**
+
+The main goal is to provide a new pod controller type, called Job, with the following characteristics:
+
+* Be able to start of one or several pods tracked as a single entity representing a one execution of this job. Final job execution status will reflect the completion status of all pods started for this run.
+
+* Ability to trigger a new job execution either on demand (triggered through server API by an external agent or by another k8 subsystem), or periodically based on a time-based  schedule defined in job specification part (using ISO8601 syntax notation).
+
+* Be able to restart a limited number of time failing containers of pods created by one job execution (using "Restart on failure" logic extended by a maximal restart count).
+
+* Be able to limit the execution time of pods created by this job.
+
+* Be able to provide status information about pending and current job executions, along with a (limited) history of previous job runs (containing start/stop times with completion exit code)
+
+* (Future evolution) Trigger jobs based on success/failure of others.
```

Comment:
OK, here's what I was referring as reaction to OnFailure and OnSuccess. I think we could provide some more information around that.